nixos/modules/services/matrix/matrix-sliding-sync.nix

```diff
@@ -90,6 +93,8 @@ in
         ExecStart = lib.getExe cfg.package;
         StateDirectory = "matrix-sliding-sync";
         WorkingDirectory = "%S/matrix-sliding-sync";
+        Restart = "on-failure";
+        RestartSec = "5s";
```

Comment:
With the default burst (5) and the default start limit interval (10s) this will never reach a failing state (see `systemd-system.conf(5)` for the global defaults and `systemd.unit(5)` for an explanantion), but will retry infinitely. I don't think that this is desirable since this effectively means that you need dedicated monitoring for this service (otherwise, checking if a unit reached the failing state was sufficient, but this won't be happening here AFAIU).

I don't have a strong opinion on whether to increase the StartLimitTimeoutSec or to decrease the restart sec here, but I'd prefer to have it implemented that way rather than having infinite retries. If somebody has an outstandingly slow Matrix setup which would regularly run into that timeout, I'd expect them to update the service's configuration anyways.

Reply:
Ah yes, that's right. I missed that. I guess we could just decrease `RestartSec` to `1s` or remove it outright to be able to reach a failing state. Do you have any preference on which to use?

Reply:
I'd say that 1s is OK (when starting up the proxy has to wait for synapse to be up anyways because of Wants= and I haven't even seen a single crash when restarting synapse itself).

This is a good step forward and an improvement over the previous state. If it turns out that this was too optimistic we can fix it later as well.

Reply:
Changed to 1s